### PR TITLE
fix(github): Better massageMarkdown implementation

### LINKED ISSUE #10902: Renovate spam GitHub issues
- [x] WhiteSource Renovate hosted app on github.com
- [ ] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [ ] Gitea
- [x] github.com
- [ ] GitHub Enterprise Server
- [ ] gitlab.com
- [ ] GitLab self-hosted

Renovate version: any

**Describe the bug**

Renovate opens pull requests linking the PRs on the changelog without any proxies. This makes the work of the maintainers harder. We can no longer easily find all the related effort to a given issue. 

**Have you created a minimal reproduction repository?**

See https://github.com/mui-org/material-ui/issues/27170

Capture d’écran 2021-07-20 à 12 34 21

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Implement `remark` plugin inspired by [remark-github](https://github.com/remarkjs/remark-github).
Type declarations aren't ideal, but any interaction with [syntax-tree](https://github.com/syntax-tree/) packages is painful since we can't install fresh ESM-only packages.

## Context:

Closes #10902

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/platform/github/__snapshots__/index.spec.ts.snap [hunk 1]
```diff
@@ -7665,7 +7665,10 @@ Array [
 ]
 `;
 
-exports[`platform/github/index massageMarkdown(input) returns updated pr body 1`] = `"https://github.com/foo/bar/issues/5 plus also [a link](https://togithub.com/foo/bar/issues/5)"`;
+exports[`platform/github/index massageMarkdown(input) returns updated pr body 1`] = `
+"[https://github.com/foo/bar/issues/5](https://togithub.com/foo/bar/issues/5) plus also [a link](https://togithub.com/foo/bar/issues/5)
```
Comment (zharinov) [inline lib/platform/github/__snapshots__/index.spec.ts.snap:7669]: 😎

**Comment by zharinov** [comment]
Example: https://github.com/renovate-testing/test-10902-markdown-github-links-1/pull/1 (open release notes, search for "our progress")

File: package.json [hunk 2]
```diff
@@ -167,6 +167,8 @@
     "luxon": "2.0.1",
     "markdown-it": "12.1.0",
     "markdown-table": "2.0.0",
+    "mdast-util-find-and-replace": "1.0.0",
+    "mdast-util-to-string": "1.0.0",
```
Comment (zharinov) [inline package.json:171]: Same versions as for `remark-github`

File: lib/platform/github/massage-markdown-links.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import findAndReplace from 'mdast-util-find-and-replace';
+import remark from 'remark';
+import type { Plugin, Transformer } from 'unified';
+// eslint-disable-next-line import/no-unresolved
```
Comment (zharinov) [inline lib/platform/github/massage-markdown-links.ts:4]: `@types/unist` exists in NPM while `unist` doesn't

Comment (viceice) [reply lib/platform/github/massage-markdown-links.ts:4]: It's a type only package 🤔

File: lib/platform/github/massage-markdown-links.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  });
+  visit(tree, 'link', (node: any) => {
+    if (node?.url) {
+      // eslint-disable-next-line no-param-reassign
+      node.url = massageLink(node.url);
+    }
+  });
+}
+
+const githubExtra: Plugin<any> = (): Transformer => transformer;
```
Comment (viceice) [inline lib/platform/github/massage-markdown-links.ts:45]: Do we really need this?

File: package.json (hunk 2 above)
Comment (viceice) [reply package.json:171]: `remark-github` seems to use `^1.0.0` which not points to some higher version, check lockfile changes below

**Review by viceice — COMMENTED** [review]
LGTM, should be bump the versions to make less lockfile changes?

File: lib/platform/github/massage-markdown-links.ts (hunk 4 above)
Comment (zharinov) [reply lib/platform/github/massage-markdown-links.ts:45]: Just didn't want to write this lambda in `.use(...)` expression

**Comment by zharinov** [comment]
Updated new dependencies to the latest version before ESM transition

**Comment by rarkins** [comment]
But are we adding dependencies now which we can't upgrade later due to ESM?

**Comment by zharinov** [comment]
Yes, it's a sad part. If I understand it right, we're making them explicit. I mean, it's just what `remark-github` used before, isn't it.

**Review by viceice — APPROVED** [review]
We've effectively the same deps as before. Just some explicit `package.json` references added.

**Comment by viceice** [comment]
@zharinov This broke some release notes

https://github.com/visualon/docker-docuum/releases/v0.20.3

image

```md
### Release Notes

<details>
<summary>visualon/docker-docuum</summary>

### [`v0.20.3`](https://togithub.com/visualon/docker-docuum/releases/v0.20.3)

[Compare Source](https://togithub.com/visualon/docker-docuum/compare/v0.20.2...v0.20.3)

See [https://github.com/stepchowfun/docuum/releases/tag/v0](https://togithub.com/stepchowfun/docuum/releases/tag/v0).20.3 for more changes

</details>
```

**Comment by zharinov** [comment]
Seems like the `remark` plugin itself worked fine, though we needed better URL-in-text recognition.
I think we should try another approach: linkify only those URLs that match `/(issue|pull|discussions)/` paths.

**Pull request merged**